### ISSUE TITLE
exception at startup when FQDN in a starter configuration

### ISSUE THREAD

**Issue by PhilLAL** [tango-controls/Astor#13]
Hi,
when launching Astor is v7.0.9 on a Debian9/TANGO9 server, an uncatched exception is launched.

Here is the astor startup message:
```
$ astor
Display is localhost:12.0
Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
	at admin.astor.tools.MySqlUtil.manageTangoHostProperties(MySqlUtil.java:231)
	at admin.astor.AstorUtil.getTangoHostList(AstorUtil.java:714)
	at admin.astor.AstorTree.initTangoObjects(AstorTree.java:375)
	at admin.astor.AstorTree.initComponent(AstorTree.java:305)
	at admin.astor.AstorTree.<init>(AstorTree.java:110)
	at admin.astor.Astor.buildTree(Astor.java:181)
	at admin.astor.Astor.<init>(Astor.java:107)
	at admin.astor.Astor$35.run(Astor.java:1361)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

Reading the device configuration in jive, I noticed a strange thing: the same server appears 2 times, on  time with fully qualifed domain (FQDN) name and the other one with simple hostname.

Moreover, FQDN device seems badly configured and makes appear following Device info:
```
Device Info
 
No IOR found in database for tango/admin/thomx-serveur2.thomx.fr
(Maybe, the device has never been exported...)
Server:          Starter/thomx-serveur2.thomx.fr
Exported:        false
last_exported:   ?
last_unexported: ?

Polling Status
 
Desc -> tango/admin/thomx-serveur2.thomx.fr Not Exported !
Reason -> TangoApi_DEVICE_NOT_EXPORTED
Origin -> Connection(tango/admin/thomx-serveur2.thomx.fr)
Desc -> Cannot import tango/admin/thomx-serveur2.thomx.fr
Reason -> TangoApi_CANNOT_IMPORT_DEVICE
Origin -> Connection.build_connection(tango/admin/thomx-serveur2.thomx.fr)
```
The simple hostname Device info is the following:
```
Device Info
 
Device:          tango/admin/thomx-serveur2
type_id:         IDL:Tango/Device_5:1.0
iiop_version:    1.2
host:            192.168.209.252 (192.168.209.252)
alternate addr.: 192.168.209.224
port:            46803
Server:          Starter/thomx-serveur2
Server PID:      13435
Exported:        true
last_exported:   26th October 2018 at 09:25:10
last_unexported: 26th October 2018 at 09:25:03

Polling Status
 
Polled command name = State
Polling period (mS) = 1000
Polling ring buffer depth = 10
Time needed for the last commands (NotifdState + HostState + RunningServers + StoppedServers + Servers + State) reading (mS) = 0.053
Data not updated since 948 mS
Delta between last records (in mS) = 1000, 999, 1000, 1000

Polled attribute name = NotifdState
Polling period (mS) = 1000
Polling ring buffer depth = 10
Time needed for the last attributes (NotifdState + HostState + RunningServers + StoppedServers + Servers + State) reading (mS) = 0.053
Data not updated since 948 mS
Delta between last records (in mS) = 1000, 999, 1000, 1000

Polled attribute name = HostState
Polling period (mS) = 1000
Polling ring buffer depth = 10
Time needed for the last attributes (NotifdState + HostState + RunningServers + StoppedServers + Servers + State) reading (mS) = 0.053
Data not updated since 948 mS
Delta between last records (in mS) = 1000, 999, 1000, 1000

Polled attribute name = RunningServers
Polling period (mS) = 1000
Polling ring buffer depth = 10
Time needed for the last attributes (NotifdState + HostState + RunningServers + StoppedServers + Servers + State) reading (mS) = 0.053
Data not updated since 948 mS
Delta between last records (in mS) = 1000, 999, 1000, 1000

Polled attribute name = StoppedServers
Polling period (mS) = 1000
Polling ring buffer depth = 10
Time needed for the last attributes (NotifdState + HostState + RunningServers + StoppedServers + Servers + State) reading (mS) = 0.053
Data not updated since 948 mS
Delta between last records (in mS) = 1000, 999, 1000, 1000

Polled attribute name = Servers
Polling period (mS) = 1000
Polling ring buffer depth = 10
Time needed for the last attributes (NotifdState + HostState + RunningServers + StoppedServers + Servers + State) reading (mS) = 0.053
Data not updated since 948 mS
Delta between last records (in mS) = 1000, 999, 1000, 1000

Polled attribute name = State
Polling period (mS) = 1000
Polling ring buffer depth = 10
Time needed for the last attributes (NotifdState + HostState + RunningServers + StoppedServers + Servers + State) reading (mS) = 0.053
Data not updated since 948 mS
Delta between last records (in mS) = 1000, 999, 1000, 1000
```

Do you have any idea how we could solve this?
I think probably /etc/hostname changed after the server installation to be completed with FQDN, and the issue begins at this time. 
However, I wonder if this issue is due to dot in the device name (see http://www.tango-controls.org/community/forum/c/general/development/characters-allowed-in-tango-deviceattribute-names/). 
If not  should it be preferable to catch this kind of exception to avoid blocking astor if one server is misconfigured on the TANGO network (ideally with a warning)? I am not an expert of Astor behaviour so I can miss something.

Regards.

**Comment by Pascal-Verdier** [comment]
Hi Philippe
I tried your case with 2 Starters, but Astor works normally.
But it is Astor-7.2.0
Of course we can start only one Starter at a time.
Did I missed something ?

![2_starters](https://user-images.githubusercontent.com/19665928/47562158-df64bc00-d91d-11e8-9697-a59a4144b74a.png)

**Comment by PhilLAL** [comment]
Ok, so this is not an issue due to 2 starter configured into Tango DB.

You don't have any error message during astor startup?

**Comment by Pascal-Verdier** [comment]
I have just on stderr:
`----------> debian8-devel.esrf.fr
`
Due to the exported info for this starter is not correct.
May be I could report a Warning in a future release.

Pascal

**Comment by PhilLAL** [comment]
I tested starting astor on a client and a server.
I think I now understand the issue: the /etc/tangorc contained a TANGO_HOST defined as IP:PORT but Starter was defined as HOSTNAME:PORT into TANGO DB.

To solve the issue, I defined TANGO_HOST to the same hostname defined into TANGO DB, restarting the starter on the computer, deleting the wrong Starter device on TANGO DB and then started astor without any issue.

```
$> export TANGO_HOST=thomx-serveur2:20000
$> astor [FAILURE]
$> sudo service tango-starter restart
$> astor [FAILURE]
$> /usr/lib/tango/tango_admin --delete-server Starter/thomx-serveur2.thomx.fr
$> astor [SUCCESS]
```

Thank you for your time Pascal, and sorry for the disturbance. 
Enjoy yourw weekend.
